tests/application-operator-tests/test/testkit/helm_client.go

```diff
@@ -1,77 +1,135 @@
 package testkit
 
 import (
-	"k8s.io/helm/pkg/helm"
-	"k8s.io/helm/pkg/proto/hapi/release"
-	rls "k8s.io/helm/pkg/proto/hapi/services"
-	"k8s.io/helm/pkg/tlsutil"
-
 	"time"
+
+	log "github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
 )
 
 type HelmClient interface {
-	CheckReleaseStatus(rlsName string) (*rls.GetReleaseStatusResponse, error)
-	CheckReleaseExistence(name string) (bool, error)
-	IsInstalled(rlsName string) bool
-	TestRelease(rlsName string) (<-chan *rls.TestReleaseResponse, <-chan error)
+	CheckReleaseStatus(rlsName string, namespace string) (release.Status, error)
+	CheckReleaseExistence(rlsName string, namespace string) (bool, error)
+	IsInstalled(rlsName string, namespace string) bool
+	TestRelease(rlsName string, namespace string) (*release.Release, error)
 }
 
 type helmClient struct {
-	helm          *helm.Client
 	retryCount    int
 	retryWaitTime time.Duration
+	helmDriver    string
+}
+
+func NewHelmClient(helmDriver string) (HelmClient, error) {
+
+	return &helmClient{
+		helmDriver: helmDriver,
+	}, nil
 }
 
-func NewHelmClient(host, tlsKeyFile, tlsCertFile string, skipVerify bool) (HelmClient, error) {
-	tlsOpts := tlsutil.Options{
-		KeyFile:            tlsKeyFile,
-		CertFile:           tlsCertFile,
-		InsecureSkipVerify: skipVerify,
+func (hc *helmClient) CheckReleaseStatus(rlsName string, namespace string) (release.Status, error) {
+
+	actionConfig, err := hc.actionConfigInit(namespace)
+	if err != nil {
+		return release.StatusUnknown, err
 	}
 
-	tlsCfg, err := tlsutil.ClientConfig(tlsOpts)
+	statusAction := action.NewStatus(actionConfig)
+	status, err := statusAction.Run(rlsName)
 	if err != nil {
-		return nil, err
+		return release.StatusUnknown, err
 	}
 
-	return &helmClient{
-		helm: helm.NewClient(helm.Host(host), helm.WithTLS(tlsCfg)),
-	}, nil
+	return status.Info.Status, nil
 }
 
-func (hc *helmClient) CheckReleaseStatus(rlsName string) (*rls.GetReleaseStatusResponse, error) {
-	return hc.helm.ReleaseStatus(rlsName)
+func (hc *helmClient) IsInstalled(rlsName string, namespace string) bool {
+
+	status, err := hc.CheckReleaseStatus(rlsName, namespace)
+
+	if err != nil {
+		return false
+	}
+	return err == nil && status == release.StatusDeployed
```

Comment:
I think only `return status == release.StatusDeployed` is needed here. The error is already checked